rudder-agent/SOURCES/patches/fusioninventory/21067-windows-updates.patch

```diff
@@ -0,0 +1,129 @@
+
+---
+ .../Inventory/Generic/Softwares/Windows.pm    | 110 ++++++++++++++++++
+ 1 file changed, 110 insertions(+)
+ create mode 100644 lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Windows.pm
+
+diff --git a/lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Windows.pm b/lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Windows.pm
+new file mode 100644
+index 0000000..3662696
+--- /dev/null
++++ b/lib/FusionInventory/Agent/Task/Inventory/Generic/Softwares/Windows.pm
+@@ -0,0 +1,110 @@
++package FusionInventory::Agent::Task::Inventory::Generic::Softwares::Windows;
++
++use strict;
++use warnings;
++
++use parent 'FusionInventory::Agent::Task::Inventory::Module';
++
++use FusionInventory::Agent::Tools;
++use JSON::PP;
++use Data::Dumper;
++
++sub isEnabled {
++    return canRun('powershell');
++}
++
++sub doInventory {
++    my (%params) = @_;
++
++    my $inventory = $params{inventory};
++    my $logger    = $params{logger};
++    my $from      = 'powershell';
++
++    my @packages = _getUpdatesList(
++        logger => $logger, from => $from
++    );
++    return unless @packages;
++
++    foreach my $package (@packages) {
++        $inventory->addEntry(
++            section => 'SOFTWAREUPDATES',
++            entry   => $package
++        );
++    }
++}
++
++sub _getUpdatesList {
++    my (%params) = @_;
++    my $logger    = $params{logger};
++    FusionInventory::Agent::Tools::Win32->use('runCommand');
++
++    # Create temp script
++    my $fh = File::Temp->new(
++        TEMPLATE    => 'get-available-updates-XXXXXX',
++        SUFFIX      => '.ps1'
++    );
++    print $fh <DATA>;
++    close($fh);
++    my $file = $fh->filename;
++    return unless ($file && -f $file);
++
++    $logger->debug("executing $file") if $logger;
++    #my $raw_json = qx($file);
++    my @lines = getAllLines(
++        command => "powershell -NonInteractive -ExecutionPolicy ByPass -File $file",
++        %params
++    );
++    my $raw_json = join(' ', @lines);
++    my @updates;
++    my @packages;
++
++    # check that it is valid JSON
++    eval {
++        my $coder = JSON::PP->new;
++        my @raw_updates = @{$coder->decode($raw_json)};
++        $logger->debug(Data::Dumper->Dump([ \@raw_updates ], [ qw(*packages) ])) if $logger;
++        @updates = @raw_updates;
++    };
++    if ($@) {
++        $logger->error("Rudder system updates detection didn't return valid JSON entry, error is:$@") if $logger;
++        my $filename = "C:/Program Files/Rudder/tmp/inventory-json-error-".time();
++        open(my $fh, '>', $filename);
++        print $fh $raw_json;
++        print $fh "\n";
++        print $fh $@;
++        close($fh);
++        $logger->error("Invalid JSON data stored in $filename") if $logger;
++    }
++    foreach my $update (@updates) {
++      my $package = {
++        NAME    => $update->{Name},
++        ARCH    => "x86_64",
++        VERSION => "1.0",
```

Comment:
if we don't have a version number we need to make it optional on the webapp side